### PR TITLE
Support not_empty function and empty function

### PR DESCRIPTION
**Pull request by Lishi-hub**
Support not_empty function 
Returns 0 for an empty string or 1 for a non-empty string.
#2593

Support empty function
Returns 1 for an empty string or 0 for a non-empty string. 
 #2592

### CODE REVIEW

File: be/src/exprs/string_functions.cpp [hunk 1]
```diff
@@ -99,6 +99,32 @@ BooleanVal StringFunctions::ends_with(
...
 }
 
+BooleanVal StringFunctions::empty(
+        FunctionContext* context, const StringVal& str) {
+    if (str.is_null) {
+        return BooleanVal::null();
+    }
+    if (str.len == 0) {
+        return 1;
+    }
```
Comment (HangyuanLiu) [inline be/src/exprs/string_functions.cpp]:
```suggestion
    } else {
```

File: be/src/exprs/string_functions.cpp [hunk 2]
```diff
@@ -99,6 +99,32 @@ BooleanVal StringFunctions::ends_with(
...
+}
+
+BooleanVal StringFunctions::not_empty(
+        FunctionContext* context, const StringVal& str) {
+    if (str.is_null) {
+        return BooleanVal::null();
+    }
+    if (str.len == 0) {
+        return 0;
+    }
```
Comment (HangyuanLiu) [inline be/src/exprs/string_functions.cpp]:
```suggestion
    } else {
```

File: be/src/exprs/string_functions.cpp [hunk 3]
```diff
@@ -99,6 +99,32 @@ BooleanVal StringFunctions::ends_with(
     return BooleanVal(str_sp.ends_with(suffix_sp));
 }
 
+BooleanVal StringFunctions::empty(
+        FunctionContext* context, const StringVal& str) {
+    if (str.is_null) {
+        return BooleanVal::null();
```
Comment (HangyuanLiu) [inline be/src/exprs/string_functions.cpp]: Considered non-empty if it contains at least one byte, even if this is a space or a null byte. The function should return true or false, not null.

File: gensrc/script/doris_builtins_functions.py [hunk 4]
```diff
@@ -508,6 +508,10 @@
         '_ZN5doris15StringFunctions9ends_withEPN9doris_udf15FunctionContextERKNS1_9StringValES6_'],
     [['starts_with'], 'BOOLEAN', ['VARCHAR', 'VARCHAR'],
         '_ZN5doris15StringFunctions11starts_withEPN9doris_udf15FunctionContextERKNS1_9StringValES6_'],
+    [['not_empty'], 'BOOLEAN', ['VARCHAR'],
+            '_ZN5doris15StringFunctions9not_emptyEPN9doris_udf15FunctionContextERKNS1_9StringValE'],
+    [['empty'], 'BOOLEAN', ['VARCHAR'],
+                    '_ZN5doris15StringFunctions5emptyEPN9doris_udf15FunctionContextERKNS1_9StringValE'],
```
Comment (HangyuanLiu) [inline gensrc/script/doris_builtins_functions.py]:
```suggestion
             '_ZN5doris15StringFunctions5emptyEPN9doris_udf15FunctionContextERKNS1_9StringValE'],
```

**Review by HangyuanLiu — CHANGES_REQUESTED** [review]

**Comment by HangyuanLiu** [comment]
Pleas Resolve confilicts

**Comment by morningman** [comment]
Is it necessary to implement this function? Why not using `length(str) == 0`?

**Comment by HangyuanLiu** [comment]
> Is it necessary to implement this function? Why not using `length(str) == 0`?
@morningman 
I think is necessary ,length(null) == null and empty(null) == 1. And empty(array) is necessary when we implement complex type

**Comment by morningman** [comment]
> I think is necessary ,length(null) == null and empty(null) == 1. And empty(array) is necessary when we implement complex type

so maybe `null_or_empty` is better? like `Strings.isNullOrEmpty(xxx)` in Guava lib ?

**Comment by HangyuanLiu** [comment]
> > I think is necessary ,length(null) == null and empty(null) == 1. And empty(array) is necessary when we implement complex type
> 
> so maybe `null_or_empty` is better? like `Strings.isNullOrEmpty(xxx)` in Guava lib ?

OK

**Pull request closed without merging**